sdcm/utils/common.py

```diff
@@ -357,11 +357,27 @@ def inner(*args, **kwargs):
                 return return_val
             return inner
 
+        results = []
         with ThreadPoolExecutor(max_workers=self.num_workers) as pool:
             LOGGER.debug("Executing in parallel: '{}' on {}".format(func.__name__, self.objects))
             if not self.disable_logging:
                 func = func_wrap(func)
-            return list(pool.map(func, self.objects, timeout=self.timeout))
+            futures = []
+
+            for obj in self.objects:
+                futures.append(pool.submit(func, obj))
+
+            for future in concurrent.futures.as_completed(futures):
+                try:
+                    result = future.result(self.timeout)
+                    results.append(result)
+                except Exception:  # pylint disable=broad-except
+                    if ignore_exception:
+                        results.append(None)
```

Comment:
missing a logger.warn or something like that, case we do want to know about those error, but not to fail cause of them.

Reply:
I think you Should put Exception Object into the results, otherwise, it is confusing(what if the function that you run is expected to return `None`?) Or you should mention this in the docstring

Reply:
fixed